packages/server/src/internals/middlewares.ts

```diff
@@ -0,0 +1,31 @@
+import { ProcedureType } from '../router';
+import { TRPCError } from '../TRPCError';
+
+export const middlewareMarker = Symbol('middlewareMarker');
+interface MiddlewareResultBase {
+  /**
+   * All middlewares should pass through their `next()`'s output.
+   * Requiring this marker makes sure that can't be forgotten at compile-time.
+   */
+  readonly marker: typeof middlewareMarker;
```

Comment:
If we're doing the union approach, maybe we don't need the `marker` prop. Just seems like overkill to have two "tag" type props.